### PR TITLE
Complete Rails 7.1 update

### PR DESCRIPTION
**Pull request by KludgeKML**
Some configuration updates wer missed when we moved from Rails 7.0 -> 7.1. This completes the update.

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: config/environments/development.rb [hunk 1]
```diff
@@ -45,20 +45,15 @@
...
   # Annotate rendered view with file names.
   # config.action_view.annotate_rendered_view_with_filenames = true
 
-  # Use an evented file watcher to asynchronously detect changes in source code,
-  # routes, locales, etc. This feature depends on the listen gem.
-  config.file_watcher = ActiveSupport::EventedFileUpdateChecker
-
-  config.hosts += [
-    "locations-api.dev.gov.uk",
-  ]
```
Comment (1pretz1) [inline config/environments/development.rb:60]: We'll need to keep the bespoke host config for local docker compatibility, as Rails will block requests for locations-api.dev.gov.uk without.

File: config/environments/production.rb [hunk 2]
```diff
@@ -15,13 +15,12 @@
...
 
-  # Ensures that a master key has been made available in either ENV["RAILS_MASTER_KEY"]
-  # or in config/master.key. This key is used to decrypt credentials (and other encrypted files).
+  # Ensures that a master key has been made available in ENV["RAILS_MASTER_KEY"], config/master.key, or an environment
+  # key such as config/credentials/production.key. This key is used to decrypt credentials (and other encrypted files).
   # config.require_master_key = true
 
-  # Disable serving static files from the `/public` folder by default since
-  # Apache or NGINX already handles this.
-  config.public_file_server.enabled = ENV["RAILS_SERVE_STATIC_FILES"].present?
```
Comment (1pretz1) [inline config/environments/production.rb:24]: This seems something we may want to keep?

**Review by 1pretz1 — COMMENTED** [review]
Nice - left a couple comments for options we may want to keep but everything else looks good!

File: config/environments/development.rb (hunk 1 above)
Comment (KludgeKML) [reply config/environments/development.rb:60]: Ah, this always catches me out. I've done a bit of digging around and I think we do want to keep the config, but not in this form - we actually want config.hosts.clear for local dev:

https://github.com/alphagov/support-api/commit/2cbd2688bdad9284d6fd38fc5c39164071331623

File: config/environments/production.rb (hunk 2 above)
Comment (1pretz1) [reply config/environments/production.rb:24]: Nice digging - if we're not configuring the ENV var for locations-api then setting to false to keep the behaviour the same seems reasonable 👍

File: config/environments/development.rb (hunk 1 above)
Comment (KludgeKML) [reply config/environments/development.rb:60]: Done

**Review by 1pretz1 — APPROVED** [review]

**Pull request merged**